rotkehlchen/balances/historical.py

```diff
@@ -99,63 +260,227 @@ def get_assets_amounts(
+                    column='asset',
+                    values=asset_identifiers,
+                    operator='IN',
+                ))
+
+            trade_where_clauses = []
+            trade_bindings: list[Any] = []
+            for filter_ in trade_filters:
+                clauses, bindings = filter_.prepare()
+                trade_where_clauses.extend(clauses)
+                trade_bindings.extend(bindings)
+
+            trades_iterator = cursor1.execute(
+                f'SELECT timestamp, id, type, base_asset, quote_asset, amount, rate '
+                f'FROM trades WHERE {" AND ".join(trade_where_clauses)} ORDER BY timestamp',
+                trade_bindings,
             )
-            main_currency = self.db.get_setting(cursor=cursor, name='main_currency')
+
+            event_where_clauses = []
+            event_bindings: list[Any] = []
+            for filter_ in event_filters:
+                clauses, bindings = filter_.prepare()
+                event_where_clauses.extend(clauses)
+                event_bindings.extend(bindings)
+
+            events_iterator = cursor2.execute(
+                f'SELECT CAST(timestamp / 1000 AS INTEGER) AS timestamp, event_identifier, asset, amount, type, subtype '  # noqa: E501
+                f'FROM history_events WHERE {" AND ".join(event_where_clauses)} ORDER BY timestamp',  # noqa: E501
+                event_bindings,
+            )
+
+            current_event = next(events_iterator, None)
+            current_trade = next(trades_iterator, None)
+            while current_event is not None or current_trade is not None:
+                try:
+                    # If there are no more trades, process the remaining events
+                    if current_trade is None:
+                        assert current_event is not None, 'current_event should never be None here'
+                        events.append(MinimalHistoryEvent.deserialize(current_event))
+                        current_event = next(events_iterator, None)
+                        continue
+
+                    # If there are no more events, process the remaining trades
+                    if current_event is None:
+                        events.append(MinimalTrade.deserialize(current_trade))
+                        current_trade = next(trades_iterator, None)
+                        continue
+
+                    # Compare timestamps and process the earlier event
+                    if current_event[0] <= current_trade[0]:
+                        events.append(MinimalHistoryEvent.deserialize(current_event))
+                        current_event = next(events_iterator, None)
+                    else:
+                        events.append(MinimalTrade.deserialize(current_trade))
+                        current_trade = next(trades_iterator, None)
+                except (UnknownAsset, DeserializationError) as e:
+                    log.error(f'Failed to deserialize event due to {e!s}')
```

Comment:
This should log also in what part of the logic you are, the name of the function or something. If you read this you wouldn't know